rtos/Thread.h

```diff
@@ -73,6 +73,15 @@ namespace rtos {
  * and underlying RTOS objects (static or dynamic RTOS memory pools are not being used).
  * Additionally the stack memory for this thread will be allocated on the heap, if it wasn't supplied to the constructor.
  */
+
+/* This flag can be used to change the default access of all threads in non-secure mode.
```

Comment:
Is there any reason to have this in the Thread header file ? Looks to me that it is only used as the default `tz_module` value used in implementation file.  

If this value is useful for users; could you move this block above the thread documentation and *doxygen* it ? 

Could you also indicates what it means if `MBED_TZ_DEFAULT_ACCESS` is set to 0 ?

Reply:
> Is there any reason to have this in the Thread header file ?

I wanted to leave the description in header file, since different target devices can use it the way they want. In case of Nuvoton few of there peripherals are in secure mode, hence they need all user threads to have secure access (specially for green-tea testing). But some other target device might keep it as no secure access for user threads.

Reply:
For compatibility and minimized impact, this is the better way for target specific setting and we could pass all green-tea & CI-Test on Nuvoton Cortex-M23 chip based on this change.